lib/workers/repository/finalise/repository-statistics.ts

```diff
@@ -33,3 +35,71 @@ export function runRenovateRepoStats(
   }
   logger.debug({ stats: prStats }, `Renovate repository PR statistics`);
 }
+
+export interface BranchMetadata {
+  branchName: string;
+  branchSha: string | null;
+  baseBranch: string | undefined;
+  baseBranchSha: string | null;
+  automerge: boolean;
+  isModified: boolean;
+}
+
+export interface BaseBranchMetadata {
+  branchName: string;
+  sha: string;
+}
+
+export interface BranchSummary {
+  cacheModified: boolean | undefined;
+  baseBranches: BaseBranchMetadata[];
+  branches: BranchMetadata[];
+  inactiveBranches: string[];
+}
+
+export function unwrap({
+  branchName,
+  sha: branchSha,
+  baseBranch,
+  parentSha: baseBranchSha,
+  automerge,
+  isModified,
+}: BranchCache): BranchMetadata {
+  return {
+    branchName,
+    branchSha,
+    baseBranch,
+    baseBranchSha,
+    automerge,
+    isModified,
+  };
+}
+
+export function runBranchSummery(): void {
+  const { scan, branches, modified: cacheModified } = getCache();
+
+  const baseMetadata: BaseBranchMetadata[] = [];
+  for (const [branchName, cached] of Object.entries(scan ?? {})) {
+    baseMetadata.push({ branchName, sha: cached.sha });
```

Comment:
Does it make sense for hashes to be in 7 characters form?
If the stats meant to be consumed by human, shorter hashes may be more practical.

@rarkins @viceice what do you think?

Reply:
sounds good

Reply:
@Gabriel-Ladzaretti does this work for us?

Reply:
Actually, I prefer to stick with the long `sha` as that's what we've been using in our backend for a long time